ydb/core/driver_lib/cli_utils/cli_cmds_tablet.cpp

```diff
@@ -83,123 +85,163 @@ class TClientCommandKeyValue : public TClientCommandTree {
     }
 };
 
-class TClientCommandTabletExec : public TClientCommandBase {
+template<class TRequest, class TResponse>
+class TClientCommandTabletCommon {
+protected:
+    int RunTabletCommon(TClientCommand::TConfig& config) {
+        grpc::ClientContext context;
+        if (auto token = AcquireSecurityToken(config)) {
+            context.AddMetadata("x-ydb-auth-ticket", *token);
+        }
+        auto channel = NYdbGrpc::CreateChannelInterface(CommandConfig.ClientConfig);
+        auto stub = Ydb::Tablet::V1::TabletService::NewStub(channel);
+        if (NClient::TKikimr::DUMP_REQUESTS) {
+            Cerr << "<-- " << TypeName<TRequest>() << "\n" << Request.DebugString();
+        }
+        auto status = Send(context, stub);
+        if (!status.ok()) {
+            Cerr << "ERROR: " << int(status.error_code()) << " " << status.error_message() << Endl;
+            return 1;
+        }
+        if (NClient::TKikimr::DUMP_REQUESTS) {
+            Cerr << "--> " << TypeName<TResponse>() << "\n" << Response.DebugString();
+        }
+        if (Response.status() != Ydb::StatusIds::SUCCESS) {
+            Cerr << "ERROR: " << Response.status() << Endl;
+            for (const auto& issue : Response.issues()) {
+                Cerr << issue.message() << Endl;
+            }
+            return 1;
+        }
+        return 0;
+    }
+
+    virtual grpc::Status Send(
+        grpc::ClientContext& context,
+        const std::unique_ptr<Ydb::Tablet::V1::TabletService::Stub>& stub) = 0;
+
+protected:
+    TRequest Request;
+    TResponse Response;
+};
+
+class TClientCommandTabletExec
+    : public TClientCommandBase
+    , public TClientCommandTabletCommon<
+        Ydb::Tablet::ExecuteTabletMiniKQLRequest,
+        Ydb::Tablet::ExecuteTabletMiniKQLResponse>
+{
 public:
     TClientCommandTabletExec()
         : TClientCommandBase("execute", { "exec" })
     {
     }
 
-    TAutoPtr<NMsgBusProxy::TBusTabletLocalMKQL> Request;
     TString Program;
-    TString Params;
 
     virtual void Config(TConfig& config) override {
         TClientCommand::Config(config);
-        config.Opts->AddLongOption("follower", "connect to follower").NoArgument();
-        config.Opts->AddLongOption("json-ui64-as-string", "json output ui64 as string").NoArgument();
-        config.Opts->AddLongOption("json-binary-as-base64", "json output binary data in base64").NoArgument();
-        config.SetFreeArgsNum(1, 2);
+        config.Opts->AddLongOption("dry-run", "test changes without applying").NoArgument();
```

Comment:
We lose backward compatibility in both execute and scheme-tx commands. Is it OK? No processes will be broken?

Reply:
These commands are for emergencies, and we expect ydbd version to be the same. But old versions will continue to work with new servers for some time.